pages/index.tsx

```diff
@@ -131,12 +131,20 @@ function index({
   }, [projectDetails]);
 
   settenant(tenant);
-  // If gift details are present set gift
-  if (giftDetails && isGift) {
+  // If gift details are present set gift (for direct gifts as it is not possible to unset them)
+  if (giftDetails && isGift && giftDetails.type === "direct") {
     setgiftDetails(giftDetails);
     setisGift(true);
   }
 
+  // If gift details are present, initialize gift (only for "invitation gifts")
+  React.useEffect(() => {
+    if (giftDetails && isGift && giftDetails.type === "invitation") {
+      setgiftDetails(giftDetails);
+      setisGift(true);
+    }
+  }, []);
```

Comment:
Does it make sense to have the code for `giftDetails.type === "direct"` directly running, but using a React hook for `giftDetails.type === "invitation"`?

Reply:
You're correct. 

The code for `giftDetails.type === "direct"` was legacy code present to counter the effects of a `useEffect` hook in `GiftForm` that changed gift `type` from `direct` to `invitation`. 

I've refactored this code and modified the `useEffect` in `GiftForm` to prevent gift type changing for `direct` gifts.